lib/buildenv/default.nix

```
@@ -0,0 +1,84 @@
+# buildEnv creates a tree of symlinks to the specified paths.  This is
+# a fork of the buildEnv in the Nix distribution.  Most changes should
+# eventually be merged back into the Nix distribution.
+
+{ buildPackages, runCommand, lib, substituteAll }:
```

For lib functions that require a systemized nixpkgs you can 
- either pass a packageset directly (thats often found in pure library flakes that don't depend on a specifically modified nixpkgs) 
- or take a system and construct the packages using the context which will ensure yoyre using the same nixpkgs as everything else.

```suggestion
{ lib }:
pkgs:
let 
    inherit (pkgs) buildPackages runCommand substituteAll;
in
...
```

```suggestion
{ context, lib }:
system:
let 
	pkgs = (context' system).nixpkgs;
    inherit (pkgs) buildPackages runCommand substituteAll;
in
...
```